pyannote/database/registry.py

```diff
@@ -0,0 +1,618 @@
+    if home_db_yml.is_file():
+        valid_paths.append(home_db_yml)
+    
+    cwd_db_yml = Path.cwd() / "database.yml"
+    if cwd_db_yml.is_file():
+        valid_paths.append(home_db_yml)
+    
+    valid_paths += _env_config_paths()
+
+    return valid_paths
+
+def get_database_yml(database_yml: Union[Text, Path] = None) -> Path:
+    """Find location of pyannote.database configuration file
+
+    Parameter
+    ---------
+    database_yml : Path, optional
+        Force using this file.
+
+    Returns
+    -------
+    path : Path
+        Path to 'database.yml'
+
+    Raises
+    ------
+    FileNotFoundError when the configuration file could not be found.
+    """
+
+    # when database_yml is provided, use it
+    if database_yml is not None:
+        database_yml = Path(database_yml).expanduser()
+        # does the provided file exist?
+        if not database_yml.is_file():
+            msg = f"File '{database_yml}' does not exist."
+            raise FileNotFoundError(msg)
+
+        return database_yml
+
+    # is there a file named "database.yml" in current working directory?
+    if (Path.cwd() / "database.yml").is_file():
+        database_yml = Path.cwd() / "database.yml"
+
+    # does PYANNOTE_DATABASE_CONFIG environment variable links to an existing file?
+    elif os.environ.get("PYANNOTE_DATABASE_CONFIG") is not None:
+        database_yml = Path(os.environ.get("PYANNOTE_DATABASE_CONFIG")).expanduser()
+        if not database_yml.is_file():
+            msg = (
+                f'"PYANNOTE_DATABASE_CONFIG" links to a file that does not'
+                f'exist: "{database_yml}".'
+            )
+            raise FileNotFoundError(msg)
+
+    # does default "~/.pyannote/database.yml" file exist?
+    else:
+        database_yml = Path("~/.pyannote/database.yml").expanduser()
+
+        # if it does not, let the user know that nothing worked and in which
+        # locations "database.yml" was looked for.
+        if not database_yml.is_file():
+            msg = (
+                f'"pyannote.database" relies on a YAML configuration file but '
+                f"could not find any. Here are the locations that were "
+                f'looked for: {Path.cwd() / "database.yml"}, {database_yml}'
+            )
+            if os.environ.get("PYANNOTE_DATABASE_CONFIG") is not None:
+                database_yml = Path(
+                    os.environ.get("PYANNOTE_DATABASE_CONFIG")
+                ).expanduser()
+                msg += (
+                    f", and {database_yml} (given by "
+                    f"PYANNOTE_DATABASE_CONFIG environment variable)."
+                )
+            else:
+                msg += "."
+            raise FileNotFoundError(msg)
+
+    return database_yml
+
+
+def _merge_protocols_inplace(new_protocols: Dict[Tuple[Text, Text], Type], old_protocols: Dict[Tuple[Text, Text], Type], allow_override:OverrideType, db_name, database_yml:str):
```

Comment:
We can make them `@staticmethods`, I left them as module-level functions because : 
- some time ago i think i read that module-level functions are the more "pythonesque" way to do it (or something..)
- and because I thought the `Registry` class should only focus on dealing with loading the yml files / shouldn't be 'aware' of it's "global" status. 
It made more sense to me to do/put everything that interacts with the "global state" (declaring the global registry variable, accessing the environment variable and user folder) in the same place (at module-level).

(so this also relates to moving the comment suggesting to move registry.load_databases to init)
(maybe it'd be easier to talk about it in person)